client/src/App.tsx

```diff
@@ -1,25 +1,53 @@
 import { createBrowserRouter, RouterProvider } from "react-router-dom";
 import Landing from "./pages/Landing.tsx";
+import Home from "./pages/Home.tsx";
+import AppShell from "./components/AppShell.tsx";
+import ErrorElement from "./pages/ErrorElement.tsx";
 
+const NAV_CONFIG = {
+  brand: "UML Mentor",
+  routes: [
+    {
+      name: "Home",
+      href: "/app/home",
+    },
+    {
+      name: "Challenges",
+      href: "/app/challenges",
+    },
+    {
+      name: "Solutions",
+      href: "/app/solutions",
+    },
+  ],
+  profile: {
+    href: "/app/profile",
+  },
+};
+
+// IMPORTANT: when adding routes to browser router, modify the NAV_CONFIG
+// accordingly.
 const router = createBrowserRouter([
   {
     path: "/",
     element: <Landing />,
-    // loader: rootLoader
-    // loaders are used to prefetch data to be used for a route
-
-    // children: [
-    //   {
-    //     path: "team",
-    //     loader: teamLoader,
-    //     element: <Team />,
-    //   },
-    // ],
+    errorElement: <ErrorElement />,
+  },
+  {
+    path: "app",
```

Comment:
why do we add an extra path? 

I'd rather have an `api` route for the server and have the client routes cleaner.

Eventually we will have the server handling initial redirects depending on session, authentication, etc. For now, you could potentially make the landing page redirect to the homepage. 

We do not have a landing page on the proposal (our main roadmap). It kinda makes sense to have a landing page tho. So that people can preview challenges without registering. We can discuss this further in the meeting. 

For now i would just get rid of the "app" route have the AppShell under `/`.

Reply:
Let's do that. The best practice I've seen is to deploy the actual app on app.<website>.com and the marketing pages like landing and blog on <website>.com